examples/pytorch/xreme-s/run_xtreme_s.py

```diff
@@ -0,0 +1,785 @@
+        else:
+            batch["labels"] = batch[data_args.target_column_name]
+        return batch
+
+    with training_args.main_process_first(desc="dataset map preprocessing"):
+        vectorized_datasets = raw_datasets.map(
+            prepare_dataset,
+            remove_columns=next(iter(raw_datasets.values())).column_names,
+            num_proc=num_workers,
+            desc="preprocess datasets",
+        )
+
+        def is_audio_in_length_range(length):
+            return length > min_input_length and length < max_input_length
+
+        # filter data that is shorter than min_input_length
+        vectorized_datasets = vectorized_datasets.filter(
+            is_audio_in_length_range,
+            num_proc=num_workers,
+            input_columns=["input_length"],
+        )
+
+    # 7. Next, we can prepare the training.
+    # Let's use word error rate (WER) as our evaluation metric,
+    # instantiate a data collator and the trainer
+
+    # Define evaluation metrics during training, *i.e.* word error rate, character error rate
+    eval_metric = load_metric("../datasets/metrics/xtreme_s", task_name)
+
+    # for large datasets it is advised to run the preprocessing on a
+    # single machine first with ``args.preprocessing_only`` since there will mostly likely
+    # be a timeout when running the script in distributed mode.
+    # In a second step ``args.preprocessing_only`` can then be set to `False` to load the
+    # cached dataset
+    if data_args.preprocessing_only:
+        logger.info(f"Data preprocessing finished. Files cached at {vectorized_datasets.cache_files}")
+        return
+
+    def compute_asr_metric(pred):
+        pred_logits = pred.predictions
+        pred_ids = np.argmax(pred_logits, axis=-1)
+
+        pred.label_ids[pred.label_ids == -100] = tokenizer.pad_token_id
+
+        pred_str = tokenizer.batch_decode(pred_ids)
+        # we do not want to group tokens when computing the metrics
+        label_str = tokenizer.batch_decode(pred.label_ids, group_tokens=False)
+
+        metric = eval_metric.compute(predictions=pred_str, references=label_str, bleu_kwargs={}, wer_kwargs={})
+        return metric
+
+    def compute_classification_metric(pred):
+        pred_ids = np.argmax(pred.predictions, axis=1)
+        metric = eval_metric.compute(predictions=pred_ids, references=pred.label_ids, bleu_kwargs={}, wer_kwargs={})
+        return metric
+
+    # Now save everything to be able to create a single processor later
+    if is_main_process(training_args.local_rank):
+        # save feature extractor, tokenizer and config
+        feature_extractor.save_pretrained(training_args.output_dir)
+        if is_text_target:
+            tokenizer.save_pretrained(training_args.output_dir)
+        config.save_pretrained(training_args.output_dir)
+
+    if is_text_target:
+        try:
+            processor = AutoProcessor.from_pretrained(training_args.output_dir)
+        except (OSError, KeyError):
```

Comment:
I'd be in favor of just removing this `except ...` statement. Every processor should now have its type saved in the config. Would make the script a bit easier to understand

Reply:
I'm confused as to where this would fail as `AutoProcessor` will use the `model_type` to determine the right processor class.